### PR TITLE
Allow manual cache purge via UI and API

### LINKED ISSUE #3962: Lookup Tables: Allow to manually clear cache
When a lookup table cache is configured with a long TTL, it can be annoying having to wait until an entry expired. A way to manually clear/invalidate the cache would be great.

### PR DESCRIPTION
**Pull request by bernd**
Also unbreak cache purging by using the data adapter id for the prefix everywhere.

Closes #3962

### CODE REVIEW

File: graylog2-server/src/main/java/org/graylog2/rest/resources/system/lookup/LookupTableResource.java [hunk 1]
```diff
@@ -209,6 +211,30 @@ public LookupResult performLookup(@ApiParam(name = "name") @PathParam("name") @N
         return lookupTableService.newBuilder().lookupTable(name).build().lookup(key);
     }
 
+    @POST
+    @Path("tables/{idOrName}/purge")
+    @ApiOperation(value = "Query a lookup table")
```
Comment (kroepke) [inline graylog2-server/src/main/java/org/graylog2/rest/resources/system/lookup/LookupTableResource.java]: The operation name is wrong

Comment (bernd) [inline graylog2-server/src/main/java/org/graylog2/rest/resources/system/lookup/LookupTableResource.java]: Fixed!

File: graylog2-server/src/main/java/org/graylog2/rest/resources/system/lookup/LookupTableResource.java [hunk 2]
```diff
@@ -209,6 +211,30 @@ public LookupResult performLookup(@ApiParam(name = "name") @PathParam("name") @N
...
 
+    @POST
+    @Path("tables/{idOrName}/purge")
+    @ApiOperation(value = "Purge lookup table cache")
+    @NoAuditEvent("Cache purge only")
+    @RequiresPermissions(RestPermissions.LOOKUP_TABLES_READ)
+    public void performPurge(@ApiParam(name = "idOrName") @PathParam("idOrName") @NotEmpty String idOrName,
+                             @ApiParam(name = "key") @QueryParam("key") String key) {
+        final Optional<LookupTableDto> lookupTableDto = dbTableService.get(idOrName);
+        if (!lookupTableDto.isPresent()) {
```
Comment (joschi) [inline graylog2-server/src/main/java/org/graylog2/rest/resources/system/lookup/LookupTableResource.java]: Drive-by-comment:
```java
LookupTableDto lookupTableDto = dbTableService.get(idOrName).orElseThrow(() -> new NotFoundException("Lookup table <" + idOrName + "> not found"));
```

File: graylog2-server/src/main/java/org/graylog2/rest/resources/system/lookup/LookupTableResource.java [hunk 3]
```diff
@@ -209,6 +211,30 @@ public LookupResult performLookup(@ApiParam(name = "name") @PathParam("name") @N
...
+            throw new NotFoundException("Lookup table <" + idOrName + "> not found");
+        }
+
+        final Optional<CachePurge> cachePurge = lookupTableService.newCachePurge(lookupTableDto.get().name());
```
Comment (joschi) [inline graylog2-server/src/main/java/org/graylog2/rest/resources/system/lookup/LookupTableResource.java]: Same here: [`Optional#orElseThrow()`](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#orElseThrow-java.util.function.Supplier-)
